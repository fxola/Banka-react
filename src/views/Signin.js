import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import HamburgerIcon from '../components/HamburgerIcon';
import Footer from '../components/Footer';
import { connect } from 'react-redux';
import { logInUserRequest } from '../actions/loginAction';
import Navigation from '../components/Navigation';

export class Signin extends Component {
  state = {
    email: '',
    password: '',
    isLoading: false
  };

  handleChange = e => {
    e.preventDefault();
    this.setState({ [e.target.name]: e.target.value });
  };

  handleSubmit = async e => {
    e.preventDefault();
    this.setState({ isLoading: true });
    await this.props.LoginUserRequest(this.state, this.props.history);
    this.setState({ isLoading: false });
  };

  render() {
    const buttonText = this.state.isLoading ? 'Loading...' : 'Log in';
    return (
      <section>
        <Navigation />
        <main id="signin-main">
          <section id="signin-section">
            <form id="signin-form" onSubmit={this.handleSubmit}>
              <input
                type="email"
                name="email"
                id="email"
                placeholder="Email"
                required="required"
                onChange={this.handleChange}
              />
              <input
                type="password"
                placeholder="password"
                name="password"
                id="password"
                required
                onChange={this.handleChange}
              />
              <button id="login-button">{buttonText}</button>
              <p>
                <Link to="#">Forgot Password ?</Link>
              </p>
              <h3>
                Don't have an account?
                <span className="signup-link">
                  <Link to="/signup">Sign Up</Link>
                </span>
              </h3>
            </form>
          </section>
        </main>
        <Footer />
      </section>
    );
  }
}
export const mapDispatchToProps = dispatch => {
  return {
    LoginUserRequest: async (userCredentials, history) => {
      return dispatch(await logInUserRequest(userCredentials, history));
    }
  };
};

export default connect(
  null,
  mapDispatchToProps
)(Signin);
