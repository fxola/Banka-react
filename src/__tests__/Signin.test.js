import React from 'react';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import Signin from '../components/Signin';
describe('Sign in Page Tests', () => {
  it('renders Sign In template', () => {
    shallow(<Signin />);
  });
});
