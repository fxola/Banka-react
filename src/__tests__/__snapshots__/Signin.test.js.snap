// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign in Components Tests renders Sign In template 1`] = `ReactWrapper {}`;
